### PR TITLE
fix(RHTAPWATCH-1042): Remove mentions of private repos (git, quay) from all namespace files (e.g. ns1)

### PR DESCRIPTION
**Pull request by Kousalya1998**
- Remove private git repositories
- fix [#53](https://github.com/konflux-ci/konflux-ci/issues/53)

### CODE REVIEW

File: test/resources/demo-users/user/managed-ns1/rpa.yaml
Comment (kousalyalakshmanan) [inline test/resources/demo-users/user/managed-ns1/rpa.yaml:1]: `"quay.io/gbenhaim0/managed/sample-component"` Not sure what should be the quay image be replaced with
@gbenhaim @yftacherzog

File: test/resources/demo-users/user/managed-ns1/rpa.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ spec:
       resolver: git
       params:
         - name: url
-          value: "https://github.com/gbenhaim/release-service-catalog.git"
+          value: "https://github.com/konflux-ci/release-service-catalog"
```
Comment (yftacherzog) [inline test/resources/demo-users/user/managed-ns1/rpa.yaml]: I don't know if both url formats are supported.

Did you confirm that this repository can be used for this purpose? Originally, those personal repos were used as the upstream ones were missing some functionality. Need to see if the functionality is there already.

File: test/resources/demo-users/user/managed-ns2/rpa.yaml [hunk 2]
```diff
@@ -23,7 +23,7 @@ spec:
       resolver: git
       params:
         - name: url
-          value: "https://github.com/gbenhaim/release-service-catalog.git"
+          value: "https://github.com/konflux-ci/release-service-catalog"
```
Comment (yftacherzog) [inline test/resources/demo-users/user/managed-ns2/rpa.yaml]: Same comment as above. 

Also, if you didn't already, please deploy your local environment and run through the steps to see that you're able to get the flow to work beginning to end. The flow uses `ns2` and `managed-ns2` manifests. Once you get it to work as-is, make the changes you have here and verify everything is still working as before.

**Review by yftacherzog — CHANGES_REQUESTED** [review]
There are still other mentions of gbenhaim in the repo. Those should be actually easier to change than the one you did.

Please go through the flow in the README (it's uses user2) and once you have it working, make the changes (to user2 manifests) and repeat the process to see it's still working.

If you get stuck in the process of deploying, let us know (slack messages or open new GitHub issues) so we can make the documentation better.

**Comment by Omeramsc** [comment]
Can you add a "fix #53" to the PR description, so the issue will be linked to the PR?

**Comment by Kousalya1998** [comment]
> There are still other mentions of gbenhaim in the repo. Those should be actually easier to change than the one you did.
> 
> Please go through the flow in the README (it's uses user2) and once you have it working, make the changes (to user2 manifests) and repeat the process to see it's still working.
> 
> If you get stuck in the process of deploying, let us know (slack messages or open new GitHub issues) so we can make the documentation better.


I was not sure what to replace the below one one with and that's why I didn't replace that
`repository: "quay.io/gbenhaim0/managed/sample-component"`

File: test/resources/demo-users/user/managed-ns1/rpa.yaml (hunk 1 above)
Comment (kousalyalakshmanan) [reply test/resources/demo-users/user/managed-ns1/rpa.yaml]: Yes, I confirmed this with Gal that day but let me check again if this is the right way to replace it

Comment (yftacherzog) [reply test/resources/demo-users/user/managed-ns1/rpa.yaml]: The way to confirm this, and the entire pr is to run through the workflow on the Readme and verify it works before the changes and then after the changes.

Comment (kousalyalakshmanan) [reply test/resources/demo-users/user/managed-ns1/rpa.yaml]: okay will check that

Comment (yftacherzog) [reply test/resources/demo-users/user/managed-ns1/rpa.yaml]: Please go through the procedure on the readme as soon as possible and let us know in slack if you run into any issues.

**Comment by yftacherzog** [comment]
> > There are still other mentions of gbenhaim in the repo. Those should be actually easier to change than the one you did.
> > 
> > Please go through the flow in the README (it's uses user2) and once you have it working, make the changes (to user2 manifests) and repeat the process to see it's still working.
> > 
> > If you get stuck in the process of deploying, let us know (slack messages or open new GitHub issues) so we can make the documentation better.
> 
> 
> Should I replace the below quay link  
> `repository: "quay.io/gbenhaim0/managed/sample-component"` with this `https://quay.io/repository/redhat-user-workloads/konflux-int-gitlab-tenant/sample-component/sample-component` ??

Were you able to run the flow as is?

File: test/resources/demo-users/user/managed-ns1/rpa.yaml (hunk 1 above)
Comment (kousalyalakshmanan) [reply test/resources/demo-users/user/managed-ns1/rpa.yaml]: I am getting this error while deploying dependencies:
```
klakshma@klakshma-thinkpadp1gen3:~/Documents/github.com/kousalya1998/konflux-ci$ ./deploy-deps.sh
Checking requirements
kubectl is installed
openssl is installed

All requirements are met
Continue
Testing PVC creation for default storage class
Creating PVC from './dependencies/pre-deployment-pvc-binding' using the cluster's default storage class
error: error validating "./dependencies/pre-deployment-pvc-binding": error validating data: failed to download openapi: Get "https://127.0.0.1:38931/openapi/v2?timeout=32s": dial tcp 127.0.0.1:38931: connect: connection refused; if you choose to ignore these errors, turn validation off with --validate=false
```

Comment (yftacherzog) [reply test/resources/demo-users/user/managed-ns1/rpa.yaml]: I'm not familiar with this error message, but google suggests it might be some sort of misalignment between the cluster and kubectl client. try updating your oc/kubectl client and kind.

Comment (yftacherzog) [reply test/resources/demo-users/user/managed-ns1/rpa.yaml]: When you find the reason for the error and the solution, please create an issues on this github to have it documented.

**Comment by Kousalya1998** [comment]
> > > There are still other mentions of gbenhaim in the repo. Those should be actually easier to change than the one you did.
> > > Please go through the flow in the README (it's uses user2) and once you have it working, make the changes (to user2 manifests) and repeat the process to see it's still working.
> > > If you get stuck in the process of deploying, let us know (slack messages or open new GitHub issues) so we can make the documentation better.
> > 
> > 
> > Should I replace the below quay link
> > `repository: "quay.io/gbenhaim0/managed/sample-component"` with this `https://quay.io/repository/redhat-user-workloads/konflux-int-gitlab-tenant/sample-component/sample-component` ??
> 
> Were you able to run the flow as is?

Yes, but I'm still seeing the same error. I verified this with Homaja, and she's experiencing a different error. I'm following the steps in the documentation and can't figure out what I might be missing.

**Comment by Kousalya1998** [comment]
@avi-biton  As we discussed yesterday in the standup, should I get rid of the ns1 stuff? Are there any updates from your end?

If yes, should I undo all the changes I made to managed-ns1/ns1 files?

File: test/resources/demo-users/user/managed-ns2/rpa.yaml [hunk 3]
```diff
@@ -7,7 +7,7 @@ metadata:
     release.appstudio.openshift.io/auto-release: 'true'
 spec:
   applications:
-    - test-component
+    - testapp
```
Comment (avi-biton) [inline test/resources/demo-users/user/managed-ns2/rpa.yaml]: Why did you change the application's name?

**Comment by avi-biton** [comment]
> @avi-biton As we discussed yesterday in the standup, should I get rid of the ns1 stuff? Are there any updates from your end?
> 
> If yes, should I undo all the changes I made to managed-ns1/ns1 files?

After discussing it with Gal, we will keep the ns1 in place.
However, when I search for `gbenhaim` in the repository, I see there are other unchanged references.
Can you please remove them all.

**Comment by Kousalya1998** [comment]
> > @avi-biton As we discussed yesterday in the standup, should I get rid of the ns1 stuff? Are there any updates from your end?
> > If yes, should I undo all the changes I made to managed-ns1/ns1 files?
> 
> After discussing it with Gal, we will keep the ns1 in place. However, when I search for `gbenhaim` in the repository, I see there are other unchanged references. Can you please remove them all?

Should I replace them with something? I dont see a public repository for sample-component.

File: test/resources/demo-users/user/managed-ns2/rpa.yaml (hunk 3 above)
Comment (kousalyalakshmanan) [reply test/resources/demo-users/user/managed-ns2/rpa.yaml]: For testing purposes, I made that change. I reverted it.

File: test/resources/demo-users/user/managed-ns1/rpa.yaml [hunk 4]
```diff
@@ -16,14 +16,14 @@ spec:
...
-          repository: "quay.io/gbenhaim0/managed/sample-component"
+          repository: "https://quay.io/repository/redhat-user-workloads/konflux-int-gitlab-tenant/sample-component/sample-component"
   origin: user-ns1
   pipeline:
     pipelineRef:
       resolver: git
       params:
         - name: url
-          value: "https://github.com/gbenhaim/release-service-catalog.git"
+          value: "https://github.com/konflux-ci/release-service-catalog.git"
```
Comment (kousalyalakshmanan) [inline test/resources/demo-users/user/managed-ns1/rpa.yaml:26]: @gbenhaim  @avi-biton PTAL at these lines that I have modified in this file. I am not sure what it needs to be replaced with

File: test/resources/demo-users/user/ns1/build-pipeline.yaml [hunk 5]
```diff
@@ -20,9 +20,9 @@ spec:
   - name: dockerfile
     value: Dockerfile
   - name: git-url
-    value: https://github.com/gbenhaim/sample-component.git
+    value: .
   - name: output-image
-    value: quay.io/gbenhaim0/sample-component:main
+    value: https://quay.io/repository/redhat-user-workloads/konflux-int-gitlab-tenant/sample-component/sample-component
```
Comment (kousalyalakshmanan) [inline test/resources/demo-users/user/ns1/build-pipeline.yaml]: @gbenhaim @avi-biton PTAL at these lines as well.

File: test/resources/demo-users/user/managed-ns1/rpa.yaml (hunk 4 above)
Comment (avi-biton) [reply test/resources/demo-users/user/managed-ns1/rpa.yaml:26]: No need to change the format of the link to quay.
But anyway, since we are not deploying this namespace, lets use a placeholder like in `managed-ns2`:
`repository: <repository url>`

File: test/resources/demo-users/user/managed-ns1/rpa.yaml [hunk 6]
```diff
@@ -16,14 +16,14 @@ spec:
     mapping:
       components:
         - name: sample-component
-          repository: "quay.io/gbenhaim0/managed/sample-component"
+          repository: "https://quay.io/repository/redhat-user-workloads/konflux-int-gitlab-tenant/sample-component/sample-component"
```
Comment (avi-biton) [inline test/resources/demo-users/user/managed-ns1/rpa.yaml]: No need to change the format of the link to quay.
But anyway, since we are not deploying this namespace, lets use a placeholder like in managed-ns2:
`repository: <repository url>`

File: test/resources/demo-users/user/ns1/build-pipeline.yaml [hunk 7]
```diff
@@ -20,9 +20,9 @@ spec:
   - name: dockerfile
     value: Dockerfile
   - name: git-url
-    value: https://github.com/gbenhaim/sample-component.git
+    value: .
```
Comment (avi-biton) [inline test/resources/demo-users/user/ns1/build-pipeline.yaml]: Lets use a placeholder here as well:
`value: <Git repository URL>`

File: test/resources/demo-users/user/ns1/build-pipeline.yaml (hunk 5 above)
Comment (avi-biton) [inline test/resources/demo-users/user/ns1/build-pipeline.yaml]:
```suggestion
    value: <Image repository URL>:tag
```

File: test/resources/demo-users/user/ns1/build-pipeline.yaml (hunk 7 above)
Comment (kousalyalakshmanan) [reply test/resources/demo-users/user/ns1/build-pipeline.yaml]: Updated in all the places.

File: test/resources/demo-users/user/ns1/build-pipeline.yaml [hunk 8]
```diff
@@ -20,9 +20,9 @@ spec:
   - name: dockerfile
     value: Dockerfile
   - name: git-url
-    value: https://github.com/gbenhaim/sample-component.git
+    value: <Git repository URl>
```
Comment (hmariset) [inline test/resources/demo-users/user/ns1/build-pipeline.yaml:23]: does this need not be replaced?

**Review by hmariset — COMMENTED** [review]
lgtm

File: test/resources/demo-users/user/ns1/build-pipeline.yaml (hunk 8 above)
Comment (avi-biton) [inline test/resources/demo-users/user/ns1/build-pipeline.yaml]:
```suggestion
    value: <Git repository URL>
```

File: test/resources/demo-users/user/ns1/build-pipeline.yaml [hunk 9]
```diff
@@ -20,9 +20,9 @@ spec:
...
   - name: output-image
-    value: quay.io/gbenhaim0/sample-component:main
+    value: <repository URL>
```
Comment (avi-biton) [inline test/resources/demo-users/user/ns1/build-pipeline.yaml]:
```suggestion
    value: <Image repository URL>
```

File: test/resources/demo-users/user/ns1/build-pipeline.yaml (hunk 8 above)
Comment (kousalyalakshmanan) [reply test/resources/demo-users/user/ns1/build-pipeline.yaml]: Updated

File: test/resources/demo-users/user/ns1/build-pipeline.yaml (hunk 9 above)
Comment (kousalyalakshmanan) [reply test/resources/demo-users/user/ns1/build-pipeline.yaml]: done

File: test/resources/demo-users/user/ns2/ec-integration-test.yaml [hunk 10]
```diff
@@ -17,7 +17,7 @@ spec:
   resolverRef:
     params:
       - name: url
-        value: 'https://github.com/gbenhaim/build-definitions'
+        value: 'https://github.com/konflux-ci/build-definitions'
       - name: revision
         value: ec-pub-key
```
Comment (yftacherzog) [inline test/resources/demo-users/user/ns2/ec-integration-test.yaml]: The revision looks strange. This should be a branch on the upstream repo, but currently  it looks to me as the name given to a pr branch on contributor's fork. Please mqnually test locally per my comment in Slack to verify.

**Review by yftacherzog — CHANGES_REQUESTED** [review]

**Comment by Kousalya1998** [comment]
> There are still other mentions of gbenhaim in the repo. Those should be actually easier to change than the one you did.
> 
> Please go through the flow in the README (it's uses user2) and once you have it working, make the changes (to user2 manifests) and repeat the process to see it's still working.
> 
> If you get stuck in the process of deploying, let us know (slack messages or open new GitHub issues) so we can make the documentation better.

I updated the PR pointing to the right branch. Now all the integration tests pass
image

File: test/resources/demo-users/user/ns2/ec-integration-test.yaml [hunk 11]
```diff
@@ -17,9 +17,9 @@ spec:
   resolverRef:
     params:
       - name: url
-        value: 'https://github.com/gbenhaim/build-definitions'
+        value: 'https://github.com/konflux-ci/build-definitions'
       - name: revision
-        value: 997f4b966aae7d65e958e63ff5b1dbdd58e4dbc6
+        value: main
```
Comment (yftacherzog) [inline test/resources/demo-users/user/ns2/ec-integration-test.yaml]: [A PR](https://github.com/konflux-ci/konflux-ci/pull/195/files#diff-f96d8721cb200706c5e4a118974dcc4fd174144c71762da0a5360268f988a649) was merged lately, changing the revision on multiple files from using branch name to using commit sha.

Check what's the latest commit on the main branch of https://github.com/konflux-ci/build-definitions and use its sha for `value` instead of `main`.

Comment (kousalyalakshmanan) [reply test/resources/demo-users/user/ns2/ec-integration-test.yaml]: @yftacherzog PTAL now. Updated it to the latest commit

**Review by amisstea — APPROVED** [review]

**Review by yftacherzog — APPROVED** [review]
lgtm
